File: 151004/Burbouski/appLab1/src/main/kotlin/com/github/hummel/dc/lab1/bean/Author.kt
package com.github.hummel.dc.lab1.bean

import com.github.hummel.dc.lab1.dto.response.AuthorResponseTo

@Suppress("MemberVisibilityCanBePrivate")
class Author(
	val id: Long, val login: String, val password: String, val firstname: String, val lastname: String
) {
	fun toResponse(): AuthorResponseTo = AuthorResponseTo(id, login, password, firstname, lastname)
}

File: 151003/Yagnnish/Lab1/Domain/Author.cs
using System.ComponentModel.DataAnnotations;

namespace lab_1.Domain
{
    public class Author:BaseEntity
    {   
        private string _login;
      
        private string _password;
        
        private string _firstname;
        
        private string _lastname;

        public Author(long? id, string login, string password, string firstname, string lastname) : base(id)
        {  
                _login = login;
                _password = password;
                _firstname = firstname;
                _lastname = lastname;
        }
        [StringLength(64,MinimumLength = 2)]
        public string Login { get => _login; }
        [StringLength(128,MinimumLength = 8)]
        
        public string Password { get => _password;}
        [StringLength(64,MinimumLength = 2)]
        public string Firstname { get => _firstname;}
        [StringLength(64,MinimumLength = 2)]
        public string Lastname { get => _lastname;}

    }
}
